system/lib/compiler-rt/lib/lsan/lsan_common_emscripten.cc

```diff
@@ -105,10 +108,60 @@ void DoStopTheWorld(StopTheWorldCallback callback, void *argument) {
   StopTheWorld(callback, argument);
 }
 
+u32 GetCurrentThread();
+
+static void ProcessThreadsCallback(ThreadContextBase *tctx, void *arg) {
```

Comment:
It looks like this was mostly copied from elsewhere. It would be good to add a comment saying whence it comes and what changes there are.

Reply:
Good point.